contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,334 @@
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageNumberOfPeriods(
+        uint8 extraNumberOfPeriods,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+        require(extraNumberOfPeriods > 0, "Invalid number of periods");
+
+        // Manage the campaign
+        ICampaignRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaignRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: extraNumberOfPeriods,
+                totalRewardAmount: 0,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit NumberOfPeriodsUpdated(extraNumberOfPeriods);
+    }
+
+    /// @notice Manage the reward per vote of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @param newMaxRewardPerVote New maximum reward per vote
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageRewardPerVote(
+        uint256 newMaxRewardPerVote,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+        require(newMaxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Manage the campaign
+        ICampaignRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaignRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: 0,
+                maxRewardPerVote: newMaxRewardPerVote
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit RewardPerVoteUpdated(newMaxRewardPerVote);
+    }
+
+    /// @notice calculate the fee amount and transfer it to the feeCollector
+    /// @return Balance after fee
+    function _handleFee() internal returns (uint256) {
+        // Cache current rewardToken balance
+        uint256 balance = IERC20(rewardToken).balanceOf(address(this));
+        require(balance > 0, "No reward to manage");
+
+        uint256 feeAmount = (balance * fee) / BASE_FEE;
+
+        // If there is a fee, transfer it to the feeCollector
+        if (feeAmount > 0) {
+            // Transfer the fee to the feeCollector
+            IERC20(rewardToken).transfer(feeCollector, feeAmount);
+
+            emit FeeCollected(feeCollector, feeAmount);
+
+            // Return the balance after fee
```

Comment:
there are tokens (like ours) which don't have a perfect up to a WEI exact accounting when doing transfers and they might be doing some rounding. It could happen that when you transfer a `feeAmount` away, that the actual balance remaining won't be `balance - feeAmount` but rather `balance - feeAmount - 1 WEI`

For that reason I would rather return `IERC20(rewardToken).balanceOf(address(this));` here

Reply:
Otherwise the whole transaction amount that was about to be transferred over to the other chain might fail

Reply:
Doing the IERC20(rewardToken).balanceOf(address(this)) is good.

Reply:
Done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2327/commits/82770e69ab5ff1e902f0d597dbd19ca0c21781d1

Reply:
I think it would be better to replace the `return balance - feeAmount` with the `IERC20(rewardToken).balanceOf(address(this));`. Since if `feeAmount == 0` then we query the `rewardsToken` contract for balance twice wasting some gas